shed/index.go

```diff
@@ -147,6 +147,37 @@ func (f Index) Get(keyFields Item) (out Item, err error) {
 	return out.Merge(keyFields), nil
 }
 
+// Fill populates fields on provided items that are part of the
+// encoded value by getting them based on information passed in their
+// fields. Every item must have all fields needed for encoding the
+// key set. The passed slice items will be changed so that they
+// contain data from the index values. No new slice is allocated.
+// This function uses a single leveldb snapshot.
+func (f Index) Fill(items []Item) (err error) {
+	snapshot, err := f.db.ldb.GetSnapshot()
+	if err != nil {
+		return err
+	}
+	defer snapshot.Release()
+
+	for i, item := range items {
+		key, err := f.encodeKeyFunc(item)
+		if err != nil {
+			return err
+		}
+		value, err := snapshot.Get(key, nil)
```

Comment:
So this assumes that all requested chunks are in the DB. Does this account for the case where GC was run in between and `value` is not found in the db? only by returning an error right? I guess this is also a feasible approach

Reply:
Yes, if the chunk is missing the error will be returned.

As the snapshot is acquired, gc will not influence this result if it runs while this function is called. The chunk will be returned in that case.

Reply:
Cool :+1: let's keep it so then. I think for now it would also make error handling more clear and easy on the calling context